### ISSUE TITLE
data loss during high ingest rate

### ISSUE THREAD

**Issue by phemmer** [influxdata/influxdb#7330]
### Bug report

__System info:__ 
influxdb 1.0
Linux & MacOS

__Steps to reproduce:__

1. Start second influxdb on port 8087
2. from `influx` shell: `create database test'`
3. from `influx` shell: `create subscription "sub0" on "test"."autogen" destinations all 'http://localhost:8087'`
4. `for ((i=0; i<30; i++)); do ( for ((n=0; n<300; n++)); do echo "insert test,foo=bar value=${i}i"; done | influx -database test ) & done; wait`
5. from `influx -database test` shell: `select sum(value) from test`
6. from `influx -database test -port 8087` shell: `select sum(value) from test` 

__Expected behavior:__
`130500`
`130500`

__Actual behavior:__
`130488`
`20188`

__Additional info:__
The first number is only sometimes off, and only by a small amount.
The second number (from the second influxdb) is always off, by a huge amount.

This is causing a problem for me as kapacitor is missing large amounts of data.

**Comment by mark-rushakoff** [comment]
If subscription writes are being dropped (which it sounds like they are in your case), you should see the `subWriteDrop` stat increase.

```
> show stats for 'write'
name: write
-----------
pointReq        pointReqHH      pointReqLocal   pointReqRemote  req     subWriteDrop    subWriteOk      writeError      writeOk writePartial    writeTimeout
265             0               265             0               33      0               33              0               33      0               0

> select subWriteDrop from _internal.."write" where time > now() - 1m
name: write
-----------
time                    subWriteDrop
2016-09-20T04:07:50Z    0
2016-09-20T04:08:00Z    0
2016-09-20T04:08:10Z    0
2016-09-20T04:08:20Z    0
2016-09-20T04:08:30Z    0
2016-09-20T04:08:40Z    0
```

You might be able to reduce subscription write drops by adjusting your batch size while keeping an eye on that stat.

**Comment by phemmer** [comment]
Negative:

```
> select subWriteDrop from _internal.."write" where time > now() - 1m
name: write
-----------
time			subWriteDrop
1474348810000000000	0
1474348820000000000	0
1474348830000000000	0
1474348840000000000	0
1474348850000000000	0
1474348860000000000	0
```

**Comment by phemmer** [comment]
Anything further on this? Data loss is kinda a big deal.

**Comment by nathanielc** [comment]
@phemmer That is definitely not expected if the subWriteDrop count is 0. Are there any errors in the InfluxDB logs?

**Comment by phemmer** [comment]
No errors.

**Comment by jwilder** [comment]
I was able to reproduce this.  `show stats for 'subscriber'` shows dropped points.  Points are getting dropped in the subscriber service here: https://github.com/influxdata/influxdb/blob/master/services/subscriber/service.go#L241

The sends into the `cw.writeRequests` channel are coming in faster than the receiver can read and process them off the channel.  The test script uses 30 concurrent writers, but the reader appears to be single goroutine: https://github.com/influxdata/influxdb/blob/master/services/subscriber/service.go#L299

**Comment by phemmer** [comment]
While I'm not sure what the fix here is going to be, can we also make the sizes of any chan buffers involved configurable. And also provide a way of reporting metrics on the sizes of those chan buffers.

**Comment by jwilder** [comment]
The fix is to increase the number of readers processing points from the channel.

**Comment by jwilder** [comment]
Fixed via #7407
